submit/submit

```diff
@@ -0,0 +1,534 @@
+#!/usr/bin/env python3
+
+#
+# submit -- command-line submit program for solutions.
+#
+# Part of the DOMjudge Programming Contest Jury System and licensed
+# under the GNU GPL. See README and COPYING for details.
+#
+
+import argparse, sys, logging, os, stat, requests, json, time
+try:
+    import magic
+except ModuleNotFoundError:
+    # Ignore, magic is optional
+    magic = None
+
+# By default no base URL is set and it is expected to use either the
+# SUBMITBASEURL environment variable or the -u/--url argument.
+# If you want to hardcode a default base URL, set it here
+baseurl = ''
+
+# Use a specific API version, set to empty string for default
+api_version = 'v4/'
+
+# Last modified time in minutes after which to warn for submitting
+# an old file.
+warn_mtime = 5
+
+num_warnings = 0
+
+def confirm(message: str) -> bool:
+    answer = ''
+    while answer not in ['y', 'n']:
+        answer = input(' (y/n) '.format(message)).lower()
+    return answer == 'y'
+
+def warn_user(msg: str):
+    global num_warnings
+    if args.quiet:
+        logging.debug('user warning #{}: {}'.format(num_warnings, msg))
+    else:
+        print('WARNING: {}!'.format(msg))
+    num_warnings += 1
+
+def usage(msg: str):
+    logging.error('error: {}'.format(msg))
+
+    print("Type '{} --help' to get help.".format(sys.argv[0]))
+
+    exit(1)
+
+def error(msg: str):
+    logging.error(msg)
+    exit(-1)
+
+def read_contests() -> list:
+    '''Read all contests from the API
+    
+    Returns:
+        The contests or None if an error occurred
+    '''
+
+    try:
+        data = do_api_request('contests')
+    except RuntimeError as e:
+        logging.warning(e)
+        return None
+
+    if not isinstance(data, list):
+        logging.warning('REST API returned unexpected JSON data for endpoint contests')
+        return None
+
+    contests = []
+
+    for contest in data:
+        if ('id' not in contest or 'shortname' not in contest or contest['id'] == ''
+            or contest['shortname'] == ''):
+            logging.warning('REST API returned unexpected JSON data for contests')
+            return None
```

Comment:
So one contest didn't contain all fields and we're not returning any data?

Reply:
This is how we did it in the C++ version. Do you think we should skip it and use the others?

Reply:
Yep

Reply:
If the contest does not contain the required data in the JSON structure something is terribly broken, right? Better to just bail than continue siliently?

Reply:
I am fine with both, but indeed it would be strange if one contest does contain the correct fields and the other does not